### ISSUE TITLE
multi user kezeles

### ISSUE THREAD

**Issue by zoltansx** [szekelymilan/e-kreta-cli#2]
Jelenleg a user config a **$HOME/.config/configstore/** mappaban az alabbi ket file-bol all:

- **e-kreta-cli.json**  - iskola, usernev, password.
- **update-notifier-e-kreta-cli.json** - last fetch timestamp(?)

Jo lenne egy megoldas, hogy tobb user (gyermek : ) ) esetén labellel (becenév) lehessen hivatkozni az egyes userekhez tartozó configra,  azaz fenti fileok-ra, mindekettő kell minden egyes usernél.

**Comment by szekelymilan** [comment]
Valóban, hasznos funkció lenne. Majd megcsinálom. 😄

Gondolkodtam rajta, hogy a `configstore` helyett alkalmazhatnék valamilyen **biztonságosabb megoldást** - jelenleg a felhasználónév és a jelszó Base64-ben van tárolva, hogy legalább egy laikus ne tudja megszerezni a bejelentkezési adatokat, de valljuk be, **ez nem profi megoldás**.
_(Hashelés nem opció, a szerverre plain text formájában kell elküldeni a felhasználónevet és a jelszót.)_

Egyébként az **update-notifier-e-kreta-cli.json** filet nem én hoztam létre, az `update-notifier` dependencyhez tartozik, ami az **automatikus frissítésért felelős.**

**Comment by zoltansx** [comment]
- _szerverre plain text formájában kell elküldeni a felhasználónevet és a jelszót_: ez tényleg challenge, mert akkor valami local vaultba (1Password, macos keychain, ...) kellene tárolni a jelszót, jó kérdés, hogy mennyire éri meg.
 - **update-notifier-e-kreta-cli.json**: valamiért azt hittem, hogy a legutolsó assignments letöltés timestampje ☺ Tényleg, a kód ellenőrzi, ha egy assignment le van töltve?

Workaround a multiuser használtara: symlinket gyártottam a **configstore** és  **assignmetns** mappákra, vsz. le is scriptelem 😄

**Comment by szekelymilan** [comment]
Valóban egy timestamp van az **update-notifier-e-kreta-cli.json** fileban, de azt az `update-notifier` dependency kezeli, nem tudom, pontosan hogyan működik.
Tetszik ez a workaround, viszont **meg lehetne rendesen is oldani**, majd kibővítem ezzel a CLI-t. 😃

**Comment by szekelymilan** [comment]
Nem, **nem ellenőrzi** a kód a letöltött házi feladatokat, minden letöltésnél **újra letölti az összeset**. Nem tudom, hogy utólag lehet-e módosítani egy kiadott feladatot a KRÉTA rendszerben, de ha igen, akkor mindenképpen ez az **előnyösebb megoldás** - különben előfordulhatna, hogy a gépeden egy régi feladatverzió marad rajta.
Azon gondolkodtam esetleg, hogy hasznos lenne-e, ha lenne egy **státuszjelző funkció** - tehát a házi feladatokat **elvégzettnek** (✔️) tudnád jelölni, ha megoldottad, és **nem elvégzettnek** (❌), ha még nem oldottad meg. Viszont itt is figyelni kéne arra, hogy **esetleg változhat a házi feladat**, és ha változott, azt **szintén jelölni** (❓) **kell**.

**Comment by zoltansx** [comment]
Az fel sem merült bennem hogy a krétában lehet esetleg utólag módosítani a feladatokat, szerintem elég valószínűtlen, hogy a tanárok ilyen funkciót kihasználnak.
Én a magam részéről a feladat állapotot amikor lesz címkézése vel oldom meg, azaz különböző színeket rendelek az egyes házifeladat fájlokhoz.
Továbbá a feladat fájlok mappáját beraktam egy git repositoryba, így tudom követni ha esetleg mégis változtak volna.

Így nekem annyi bőven elég lenne, ha meglevő fájt a kréta kliens nem törölné le csak a tartalmát irná át.

**Comment by szekelymilan** [comment]
Nem törli le a feladatokat a kliens, csak felülírja a fileokat.

**Comment by zoltansx** [comment]
Szuper, akkor tud annyit, amennyit én a Krétából ki szeretnék hozni 😄

**Comment by zoltansx** [comment]
Ld. fent, lescripteltem a multi-user kezelést symlinkekkel, jól műxik.

**Comment by szekelymilan** [comment]
**Rendes megoldást** is szeretnék majd erre készíteni a jövőben, amihez nem kell **egyénileg trükközni**. 😄
